pyquil/tests/test_qpu.py

```diff
@@ -248,5 +249,5 @@ def test_run_expects_executable(qvm, qpu_compiler):
 
 
 def test_qpu_not_engaged_error():
-    with pytest.raises(RuntimeError):
-        qpu = QPU(None)
+    with pytest.raises(UserMessageError):
+        qpu = QPU().build_client()
```

Comment:
this is what im talking about -- looks like behavior is changing

Reply:
question remains

Reply:
from offline chat, no question was apparent to me, but if you mean that this should be in the constructor, can do. Per a conversation a few weeks back I understood we'd want to leave engagement to happen lazily on the first need for the client, rather than up-front on `get_qc`, for situations in which those might have a meaningful interval in between. Thoughts?